### ISSUE TITLE
Problem reading PLY files with only positions and normals (on Windows 10)

### ISSUE THREAD

**Issue by diegonzalezb** [nmellado/Super4PCS#13]
Hello:

I'm trying to use your project for reading two very simple PLY files that have only the info. of the positions and normals, but the code fails when trying to read these files.
The specific error is in the line No. 338 of the file io_ply.h:
`if (feof (in)){
        cerr << "(PLY) incomplete file" << endl;
        return false;
    }`

I believe that these lines of code should just not be there: if the PLY file does not contain any kind of facets information, the code will fail, or perhaps I am missing something?.

The PLY files that I'm using are in ASCII format: I am attaching here an example of two of the files that I am trying to read.
[PlyExamples.zip](https://github.com/nmellado/Super4PCS/files/284603/PlyExamples.zip)

If you could please let me know if these files have something missing or wrong for me to use them with your code, I will much appreciate it...

On the other hand, I have doubts about the parameters for PLY files like the ones that I am trying to use, which have very few information, (they have very few points). 
In particular, in this case, the "delta" parameter: I think I could try a very small value, (0.01 or less)?. 
Also, for the "number of sampled points", if I want to use many of the points in the PLY files, can I set this parameter to a big value, (for example: 400 points)? ... However, if the PLY files contain less than this number of points, will this cause a problem for the algorithm?: I mean, if the actual number of points is less than this parameter, the algorithm could have some problem?

Thank you so, so much in advance for your help,

Diego.

**Comment by nmellado** [comment]
Hi,

I've added guards to skip the tests when the face list is empty (0c74d2a).

The geometry you sent is quite simple, so you don't need so much points.
See below an example (command+associated result).

Note that there is an ambiguity here, along the axis of the pipe: the algorithm cannot know, only from the geometry, how to slide the two pipes. It chooses the solution that maximize the overlap.

> Super4PCS -i seg5.ply seg13.ply -d 0.01 -o 0.95 -n 200 -t 1000

![snapshot00](https://cloud.githubusercontent.com/assets/6310221/15579389/baee95ba-2364-11e6-817e-578b5146b2c8.png)
![snapshot01](https://cloud.githubusercontent.com/assets/6310221/15579418/d0144796-2364-11e6-952f-b32eba0c5b15.png)

**Comment by nmellado** [comment]
Hi @diegonzalezb,

could you please confirm me the issue is fixed, so I can close it ?

Thanks

**Comment by diegonzalezb** [comment]
Hello @nmellado:

I'm very sorry for my very, very late reply. I've been extremely busy trying to complete other parts of my research...

Anyway, I tried the code with the parameters that you suggested:
"Super4PCS -i seg5.ply seg13.ply -d 0.01 -o 0.95 -n 200 -t 1000"

I found problems when trying to generate the Output PLY file. (I think it is basically due to issue #16, since I'm working also on Windows). 
However, I did changes in the "WritePly" function, (just to see if it worked fine). The change that I tried to do was writing only the coordinates, (without normals or color info), in plain ASCII format, (without any binary coding).
However, now I have found another problem: **the algorithm is taking too long to generate a solution**: for two tests that I did, one of them finished after more than 24 hours, and also, the result was not good: the generated object was "broken", (as if the model had several missing parts).
It is possible that I had made a mistake when changing the "WritePly" code and that is causing the "missing parts" issue, but the problem of the speed is not related to this, and I really don't know what could be causing that very slow behavior of the algorithm.

So, to summarize, the original fix that you did worked okay, but both the issue #16, and the speed problem has prevented me from making more progress in my tests.
I guess I should better open a new issue about the speed problem, correct?

Thanks a lot again @nmellado.
(I'll make sure to answer much more promptly from now on, in case you need any feedback),

Regards, 

Diego

**Comment by nmellado** [comment]
Hi,

Issue #16 is still open, I've been busy these days in a conference, I'll 
try to fix that next week.
Sorry for that.

The timing problems you are having seems to be, indeed, another problem.
Are these running time occurring for the command line you've written at 
the top of your post ?
I would say that super long timings are caused by improper input 
parameters or, at least, too restrictives.
If relevant, can you provide me with the 3d models + parameters so I can 
check what is going ?

Cheers

**Comment by diegonzalezb** [comment]
Hello again @nmellado:

Since the issue #16 is still open, for the moment I'm just trying to obtain the Transformation Matrix using the program.
Currently I am running the program with the following parameters:
"Super4PCS -i seg2.ply seg1.ply -d 0.01 -o 0.95 -n 100 -t 1000 -c 1e9"

The number of sampled points (n), I had to reduce it in some cases, because I'm testing with very small objects, but I have also made tests with larger objects, using n= 200.
Using this command line, I was able to run the algorithm much faster, (it is taking approx 30 seconds).
I believe that this is still a large time considering the small size of the objects that I am using. (I am attaching in the ZIP file, the two PLY files that I am using)
[testPLYs.zip](https://github.com/nmellado/Super4PCS/files/359729/testPLYs.zip)

However, the time is not such a big issue right now, but I would like to know if you could please let me know if I should use different parameters, so that I can obtain a correct Transformation. **As you can see in the JPG file that I am attaching, the Transformation Matrix is never changed, (it remains as the Identity), so it looks like I am using the parameters in a very wrong way.**
![super4pcs_screen](https://cloud.githubusercontent.com/assets/15080655/16773521/487f466e-4826-11e6-9578-e4541c453ccd.jpg)

Or perhaps, there is some restriction with the algorithm for small objects like the ones that I am using right now?.

Any suggestion on how to fix this problem will be much appreciated.

Thank you so much,

Diego

**Comment by nmellado** [comment]
Hi @diegonzalezb ,

Sorry for the delay I'm not able to work on the code this days.
Hopefully I might be able to come back to you next week.

Cheers.
